src/routes/url.ts

```diff
@@ -60,7 +61,59 @@ const retrieveUrl: Route<{ Params: { id: string } }> = {
 	},
 }
 
+/* Update URL origin path */
+const updateUrl: Route<{ Params: { id: string }; Body: { url: string } }> = {
+	method: 'PUT',
+	url: '/url/:id',
+	schema: {
+		body: {
+			type: 'object',
+			required: ['url'],
+			properties: {
+				url: { type: 'string' },
+			},
+		},
+	},
+	attachValidation: true,
+	async handler(request) {
+		await validateUrl(request.body.url)
+
+		const storedUrl = await this.storage.url.edit(request.params.id, request.body.url)
+		if (typeof storedUrl === 'undefined') throw new NotFoundError()
+
+		return storedUrl
+	},
+}
+
+/* Delete URL from store  */
+const deleteUrl: Route<{ Params: { id: string } }> = {
+	method: 'DELETE',
+	url: '/url/:id',
+	schema: {
+		params: {
+			type: 'object',
+			required: ['id'],
+			properties: {
+				id: { type: 'string' },
+			},
+		},
+	},
+	async handler(request, repl) {
+		const isAuth = await this.auth.isAuthorized(request)
+		if (!isAuth) throw new UnauthorizedError()
+		try {
+			await this.storage.url.delete(request.params.id)
+		} catch (e) {
+			this.log.warn('delete url failed in deleteUrl endpoint')
+			this.log.error(e.message)
+		}
+		return repl.code(200).send()
```

Comment:
because operation succeeded

Reply:
You don't need it. It will return 200 even without that line.

Also, if the operation failed, we need to return an error indicating it in the catch block.

Reply:
Done. I also throw GeneralError in the catch block

Reply:
I returned `.send(200)` because we can’t return `undefined` in the promise so we get a  `FST_ERR_PROMISE_NOT_FULLFILLED`
https://github.com/fastify/fastify/blob/main/docs/Routes.md#promise-resolution
so what do you what me to return on success? Promise of 'success' str?